File: rna/c/rna.c
#include <stdio.h>

int main()
{
    char dna_string[1000];

    scanf("%s", dna_string);

    printf("Analyzing:\n%s\n\n", dna_string);

    int i = 0;
    char curr_base = *(dna_string + i);

    while(i <= 1000 && curr_base != '\0')
    {
        if (curr_base == 'T')
        {
            dna_string[i] = 'U';
        }

        ++i;
        curr_base = *(dna_string + i);
    }

    printf("Output:\n%s\n", dna_string);

    return 0;
}

File: rna/elixir/rna.exs
defmodule RNA do
  def convert_to_rna(string) do
    String.graphemes(string) |>
    Enum.map(
      fn(x) ->
        case x do
          "T" -> "U"
          "\n" -> ""
          _ -> x
        end
      end) |>
    Enum.reduce(fn(x, acc) -> acc <> x end)
  end
end

##########
#  MAIN  #
##########

dna_string = IO.gets("")

IO.puts("Analyzing:")
IO.puts(dna_string)

IO.puts("Result:")
IO.puts(RNA.convert_to_rna(dna_string))
